### ISSUE TITLE
	software.amazon.awscdk.services.apigateway.SpecRestApi: adjust method to add integration

### ISSUE THREAD

**Issue by argenstijn** [aws/aws-cdk#29604]
### Describe the feature

I would like to create an API by an openapi schema. But i want to change one method of the API to add an integration to it. I don't want to change the openapi schema (using AWS tags to configure the integration!).

Currently this is impossible to do? right? 

### Use Case

I want to add the integration on POST method in CDK itself.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

latest

### Environment details (OS name and version, etc.)

linux

**Comment by argenstijn** [comment]
Or it would be nice to change the openapi schema in CDK. But then it would be very nice that AWSIntegration could be represented as OPEN API schema integration:

    "x-amazon-apigateway-integration" : {
          "credentials" : "arn:aws:iam::xxxxxxxxxxxxxxx:role/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
          "httpMethod" : "POST",
          "uri" : "arn:aws:apigateway:yyyyyy:events:action/PutEvents",
          "responses" : {
            "default" : {
              "statusCode" : "200",
              "responseParameters" : {
                "method.response.header.Access-Control-Allow-Methods" : "'OPTIONS,POST'",
                "method.response.header.Access-Control-Allow-Headers" : "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'",
                "method.response.header.Access-Control-Allow-Origin" : "'*'"
              },
              "responseTemplates" : {
                "application/json" : "#set($val = $input.path('$.Entries[0].EventId'))\n{\n  \"eventId\": \"$val\"\n}"
              }
            }
          },
          "requestTemplates" : {
            "application/json" : "#set($context.requestOverride.header.X-Amz-Target = \"AWSEvents.PutEvents\")\n#set($context.requestOverride.header.Content-Type = \"application/x-amz-json-1.1\")            \n#set($inputRoot = $input.json('$')) \n{ \n  \"Entries\": [\n    {\n      \"Detail\": \"$util.escapeJavaScript($inputRoot).replaceAll(\"\\\\'\",\"'\")\",\n      \"DetailType\": \"ddddddddddddddddddddd\",\n      \"EventBusName\": \"xxxxxxxxxxxxxxxxxxxxxxxxx\",\n      \"Source\":\"AABRelatieMutationPublication\"\n    }\n  ]\n}"
          },
          "passthroughBehavior" : "never",
          "type" : "aws"
        }
      }

**Comment by tim-finnigan** [comment]
Thanks for the feature request. Linking related documentation for reference: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.SpecRestApi.html

As mentioned there:
> All Resources and Methods need to be defined as part of the OpenAPI specification file, and cannot be added via the CDK.

So I'm not sure this feature request could be considered, unless I'm missing or misunderstanding something.

**Comment by argenstijn** [comment]
@tim-finnigan Yes, you cannot add. But my question is not about adding but about changed.

So for example: through openapi schema you define a method X for URI Y. With CDK you only change/add an integration to the method X for URI Y.

If that is not possible it would be nice if you could define the integration within CDK and you could have a openapi schema json/yaml representation of that. That way you can inject this part into the openapi schema before creating the SpecRestApi.
